prow/tide/status.go

```diff
@@ -414,7 +415,7 @@ func (sc *statusController) setStatuses(all []CodeReviewCommon, pool map[string]
 	process := func(pr *CodeReviewCommon) {
 		processed.Insert(prKey(pr))
 		log := sc.logger.WithFields(pr.logFields())
-		contexts, err := headContexts(log, sc.ghc, pr)
+		contexts, err := (&GitHubInteractor{ghc: sc.ghc}).headContexts(pr)
```

Comment:
Why do we construct this in the status controller on demand rather than storing it on the status controller struct?

Also I am very sure this needs `usesAppsAuth` to be set correctly

Reply:
good catch, done